streaming/base/sampling.py

```diff
@@ -23,6 +23,18 @@ def get_sampling(samples_per_shard: NDArray[np.int64], choose: int, granularity:
     Returns:
         NDArray[np.int64]: Array of ephemeral samples chosen per shard.
     """
+    if choose < 0:
+        raise ValueError('Choose must be a non-negative integer.')
+
+    if granularity <= 0:
```

Comment:
Can you please add this check in dataset.py since it is a user args and name here is confusing in comparison to `sampling_granularity`?

Reply:
Should the boundary check be: granularity <=0 or granularity > samples_per_shard * choose // num_samples?

Reply:
@xiaohanzhan-db can you explain the second option? The way it's implemented, sampling granularity larger than samples per shard is fine; it will just eat the rest of the shard.

Reply:
@karan6181

> Can you please add this check in dataset.py since it is a user args

check added in dataset.py, nice catch

> name here is confusing in comparison to sampling_granularity?

Without having the time to go and verify this has been done uniformly in everywhere, this is actually following the naming convention that we have been using. Also, there is no other kind of granularity in streaming.

Said naming convention:
- StreamingDataset has groups of args to configure the different stages of epoch generation (i.e., sampling, partitioning, shuffling, and now batching)
- In StreamingDataset, these args' names are prefixed with what stage they belong to (because we have a lot of args)
- In the stages themselves, the module and the method are already named after the stage, so we drop the redundant arg name stage prefix